src/Smalot/PdfParser/RawData/RawDataParser.php

```diff
@@ -796,6 +804,39 @@ protected function getRawObject(string $pdfData, int $offset = 0): array
         return [$objtype, $objval, $offset];
     }
 
+    /**
+     * Get value of an object header's section
+     *
+     * @param string            $key     header's section name
+     * @param string            $type    type of the section (i.e. 'numeric', '/', '<<', etc.)
+     * @param string|array|null $default default value for header's section
+     *
+     * @return string|array|null value of obj header's section, or default value if none found, or its type doesn't match $type param
+     */
+    private function getHeaderValue(?array $headerDic, string $key, string $type, $default = '')
+    {
+        if (!\is_array($headerDic)) {
```

Comment:
```suggestion
        if (false === \is_array($headerDic)) {
```

Does it make sense to you to expand this by including a check that `$headerDic` is not empty?

Reply:
I don't see any sense to add this empty check. 
If it's empty, foreach will just do nothing and function will return $default at the last line.

P.S. A comment in [doc](https://www.php.net/manual/en/language.operators.logical.php#90386):
> Evaluation of logical expressions is stopped as soon as the result is known.

It's too powerful way of optimization to ignore it